include/fastdds/rtps/reader/RTPSReader.h

```diff
@@ -43,18 +48,22 @@ struct CacheChange_t;
 struct ReaderHistoryState;
 class WriterProxyData;
 class IDataSharingListener;
+class RTPSMessageGroup;
 
 /**
  * Class RTPSReader, manages the reception of data from its matched writers.
  * @ingroup READER_MODULE
  */
-class RTPSReader : public Endpoint
+class RTPSReader
+    : public Endpoint
+    , public fastdds::statistics::StatisticsReaderImpl
```

Comment:
The key issue here is that we are *instrumenting* the actual implementation of reader, writer, and participant entities.

In the original design, the goal was to introduce the callbacks in the actual entities code seamlessly, that is, on release without the *FASTDDS_STATISTICS* no machine code is generated for them. This can only be achieved by making the real entities inherit from statistics base classes that provide the new APIs (empty or not depending on the flag).

Is similar to the security implementation on those very entities, but in the security case, no inheritance but preprocessor magic was used. Using preprocessor directives led to many headaches because slipping one of the conditional branches was very common, thus it was favored the inheritance approach.